src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp

```diff
@@ -276,8 +351,95 @@ std::any SubstraitPlanRelationVisitor::visitRelation(
       finally([&]() { currentRelationScope_ = previousScope; });
   currentRelationScope_ = symbol;
 
-  visitChildren(ctx);
+  addInputFieldsToSchema(relationType, relationData);
+
+  // Visit everything but the emit details to gather necessary information.
+  for (auto detail : ctx->relation_detail()) {
+    if (!isRelationEmitDetail(detail)) {
+      visitRelationDetail(detail);
+    }
+  }
+
+  addExpressionsToSchema(relationData);
+
+  // Now visit the emit details.
+  for (auto detail : ctx->relation_detail()) {
+    if (isRelationEmitDetail(detail)) {
+      visitRelationDetail(detail);
+    }
+  }
+
+  // Aggregate relations are different in that they alter the emitted fields
+  // by default.
+  if (relationType == RelationType::kAggregate) {
```

Comment:
Should we add a future PR or TODO for supporting aggregates with emit?

Reply:
Added to TODO at line 1994 (where the error is raised when this occurs).